### PR TITLE
Add more articles

### PR DESCRIPTION
**Pull request by MortalFlesh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,17 +18,24 @@ A curated list of amazingly awesome resources for (not only) LMC developers.
     - [Why write tests](https://www.ibuildings.nl/blog/2016/08/why-write-tests)
 - [SOLID, GRASP](https://www.zdrojak.cz/serialy/principy-objektove-orientovaneho-navrhu/)
 - [Verzování Databáze - Martin Major](https://www.youtube.com/watch?v=KTmlw5AKM8E)
+- [Technický dluh](http://blog.think-forth.com/2016/01/21/technicky-dluh/)
 
 ### PHP
 - [Symfony komponenty](https://www.zdrojak.cz/serialy/symfony-po-kruckach/)
+- [Fractal of bad design](http://eev.ee/blog/2012/04/09/php-a-fractal-of-bad-design/)
```
Comment (OndraM) [inline README.md:25]: I know this article, it *was* interesting, but IMHO article from 2012 (PHP 5.4 was the newest then)  is no longer relevant nowadays (and maybe adds much more confusion - "There is no finally construct", "PHP errors cannot be caught with try/catch", "Array unpacking can be done with the list($a, $b) " and like half of the article is outdated and wrong for PHP 7.1+).

File: README.md [hunk 2]
```diff
@@ -18,17 +18,24 @@ A curated list of amazingly awesome resources for (not only) LMC developers.
...
+- [Formatting exceptions](http://rosstuck.com/formatting-exception-messages/)
```
Comment (OndraM) [inline README.md:26]: I know another nice article about this topic (exceptions in PHP), could you please remove this from the PR and we can have have a separate discussion elsewhere (maybe create in issue and wa can post our suggestions there)? So we don't add both of them, as they may cover the same topic.

File: README.md [hunk 3]
```diff
@@ -18,17 +18,24 @@ A curated list of amazingly awesome resources for (not only) LMC developers.
...
+- [Doctrine hydratation optimization](https://ocramius.github.io/blog/doctrine-orm-optimization-hydration/)
```
Comment (OndraM) [inline README.md:27]: I'm not saying this is not interesting theory (it is!), but I'm not convinced this should be part of "articles essential to read" - this is very specialized topic about Doctrine internals.

File: README.md [hunk 4]
```diff
@@ -18,17 +18,24 @@ A curated list of amazingly awesome resources for (not only) LMC developers.
...
 
 ## Code review
 - [Code reviews v praxi](https://www.zdrojak.cz/clanky/code-reviews-praxi/])
 - [Code review checklist](https://www.zdrojak.cz/clanky/code-review-checklist/)
 - [Co nás naučilo code review](https://www.zdrojak.cz/clanky/co-nas-naucilo-code-review/)
+- [Giving better code review](https://medium.com/@mrjoelkemp/giving-better-code-reviews-16109e0fdd36)
 
 ## Q&A
 - [Funkční systémové testování: když unit-testy nestačí - @OndraM](http://prvnivcesku.cz/systemove-testovani/)
+- [PHPUnit - best practices](https://thephp.cc/news/2016/02/questioning-phpunit-best-practices)
```
Comment (OndraM) [inline README.md:38]: IMO this is no longer and issue - the old syntax was removed from new PHPUnit and [codestyle](https://github.com/lmc-eu/php-coding-standard/) forbids the annotations. So I think we are already covered :).

File: README.md [hunk 5]
```diff
@@ -18,17 +18,24 @@ A curated list of amazingly awesome resources for (not only) LMC developers.
...
+- [Fractal of bad design](http://eev.ee/blog/2012/04/09/php-a-fractal-of-bad-design/)
+- [Formatting exceptions](http://rosstuck.com/formatting-exception-messages/)
+- [Doctrine hydratation optimization](https://ocramius.github.io/blog/doctrine-orm-optimization-hydration/)
+- [Functional programming](https://speakerdeck.com/pwm/functional-programming-in-php)
 
 ## Code review
 - [Code reviews v praxi](https://www.zdrojak.cz/clanky/code-reviews-praxi/])
 - [Code review checklist](https://www.zdrojak.cz/clanky/code-review-checklist/)
 - [Co nás naučilo code review](https://www.zdrojak.cz/clanky/co-nas-naucilo-code-review/)
+- [Giving better code review](https://medium.com/@mrjoelkemp/giving-better-code-reviews-16109e0fdd36)
```
Comment (OndraM) [inline README.md:34]: Dtto, I suggest putting this to separate PR, maybe we can find other CR-related articles and we could include only the "best" one.

File: README.md [hunk 6]
```diff
@@ -18,17 +18,24 @@ A curated list of amazingly awesome resources for (not only) LMC developers.
     - [Why write tests](https://www.ibuildings.nl/blog/2016/08/why-write-tests)
 - [SOLID, GRASP](https://www.zdrojak.cz/serialy/principy-objektove-orientovaneho-navrhu/)
 - [Verzování Databáze - Martin Major](https://www.youtube.com/watch?v=KTmlw5AKM8E)
+- [Technický dluh](http://blog.think-forth.com/2016/01/21/technicky-dluh/)
```
Comment (OndraM) [inline README.md:21]: Nice one, I didn't know it, it sums the nice things :+1: .

File: README.md [hunk 7]
```diff
@@ -18,17 +18,24 @@ A curated list of amazingly awesome resources for (not only) LMC developers.
...
 
 ### PHP
 - [Symfony komponenty](https://www.zdrojak.cz/serialy/symfony-po-kruckach/)
+- [Fractal of bad design](http://eev.ee/blog/2012/04/09/php-a-fractal-of-bad-design/)
+- [Formatting exceptions](http://rosstuck.com/formatting-exception-messages/)
+- [Doctrine hydratation optimization](https://ocramius.github.io/blog/doctrine-orm-optimization-hydration/)
+- [Functional programming](https://speakerdeck.com/pwm/functional-programming-in-php)
```
Comment (OndraM) [inline README.md:28]: Also a nice topic. But don't you think this is very "pro" one? I'm not sure its worth it to confuse newcommers and junior-devs with this concepts, and PHP is not functional language, so "FP in PHP" is by definition not an essential knowledge.

But I think we may add something more introductory to basic FP concepts. Don't you know some interesting (and ideally short) article about this?

**Review by OndraM — CHANGES_REQUESTED** [review]

**Comment by MortalFlesh** [comment]
ok i will send it separately by article

**Pull request closed without merging**